### DISCUSSION TITLE
[5.x] FR: User Preference for Sidebar & Details Pane Default Toggle States

### DISCUSSION THREAD

**Discussion by richhayler** [craftcms/cms#14837] [Ideas]
Having the option to toggle the “main nav sidebar” and the “details pane” is great for editing entries on a narrow screen - e.g. monitor in portrait mode, but they aren’t persistent. 

It would be really great if the default toggle states for these could be set as a user preference option.

For example, adding checkbox options under “general settings” as:
- Collapse main navigation
- Collapse details pane

![IMG_4996](https://github.com/craftcms/cms/assets/2505643/ff648b7f-5b1d-4c11-981c-e6dfbcf61839)

**Comment by brandonkelly** [comment]
Th main sidebar view state will persist across pages/session, so there’s no point in adding a preference for that.

**Comment by richhayler** [comment]
Hey @brandonkelly, yes that is totally true. I didn't phase my suggestion correctly and should have clarified to make this persistent based on user login - i.e. the user preference would apply whether you logged into a different machine, accessed the control panel via a new private browser window or cleared session data.

Currently it's easy enough to toggle the sidebar main nav once logged in and have that preference apply for the duration of the session as you explained.

To make the details pane toggle state apply for the duration of a user session may not be everyone's preference. So I suggested the default toggle state could be set as a logged in user preference - i.e. it works in reverse to the way it does now. So if set to "collapse by default" then it would allow the user to toggle to "expand" (on a page with the details pane), but then return to "collapsed" when navigating to another page.

For consistency I suggested to have both the sidebar and details pane default states to be a "settable" user preference.
